### DISCUSSION TITLE
Optimizing Restraints

### DISCUSSION THREAD

**Discussion by Dan-Burns** [chaidiscovery/chai-lab#265] [General]
Hello,

Loving Chai - very grateful to you guys for developing it and making open source!

I'd like your thoughts on how to best use restraints - which is a really exciting enhancement.

I've been trying to force a structure to adopt as much of the available experimental conformation as possible using restraints. I take the output from a contact calculation and can have precise distances and residue specifications. 

In one case, this has worked pretty bad, possibly because of a mismatch in my residue IDs or perhaps because I'm providing too many restraints. In another case and a different protein, I have only a half dozen specified interactions and generous min and max distances and it looks like it's improved the output.

Is it better to have fewer restraints and bigger distance specifications?

Looking forward to an implementation of the confidence score.

Thanks!

Dan

**Comment by arogozhnikov** [comment]
I don't see specific reason why more restraints would be worse - model should be capable of using this information; 
but good answer here is 'try and see' - you can subsample restrains and see if that works better. 

Note all remarks about the format - it is future looking, and includes things that model can't support (e.g. min distance, confidence in restrain) - i.e. we develop format so that future models could use it without adjustments.
